### PR TITLE
Add seed kwarg to the tutorial scatter dataset

### PR DESCRIPTION
**Pull request by Illviljan**
Allow controlling the randomness of the dataset. It's difficult to catch issues with the dataset if it always changes each run.

### CODE REVIEW

**Comment by Illviljan** [comment]
Wasn't this dataset that was the root cause for my issues, was a rogue set() that changed orders...
I see now that the values that are randomized are converted to strings anyway so it doesn't matter if it's random or not.

We can merge this anyway the mypy issues was real and using the default_rng follows numpys recommendations better.

File: xarray/tutorial.py [hunk 1]
```diff
@@ -226,15 +226,24 @@ def load_dataset(*args, **kwargs):
...
         dims=["x", "y", "z", "w"],
         coords=[
-            np.arange(3),
-            np.linspace(0, 1, 11),
-            np.arange(4),
-            0.1 * np.random.randn(4),
+            ("x", np.arange(3)),
+            ("y", np.linspace(0, 1, 11)),
+            ("z", np.arange(4)),
+            ("w", 0.1 * rng.standard_normal(4)),
```
Comment (keewis) [inline xarray/tutorial.py:246]: it looks like this `list` of key-value `tuple`s should just become a `dict`...

**Review by keewis — COMMENTED** [review]
two suggestions, but otherwise I agree: we should be able to merge

File: xarray/tutorial.py (hunk 1 above)
Comment (Illviljan) [reply xarray/tutorial.py]: Done.

File: xarray/tutorial.py [hunk 2]
```diff
@@ -226,15 +226,24 @@ def load_dataset(*args, **kwargs):
...
 
-def scatter_example_dataset():
+def scatter_example_dataset(*, seed=None) -> Dataset:
+    """
+    Create an example dataset.
+
+    Parameters
+    ----------
+    seed : integer, optional
+        Seed for the random number generation. The default is None.
```
[the remark this answers is not in the document]
Comment (Illviljan) [reply xarray/tutorial.py]: Done, I just took whatever Spyder does when it generates docstrings.

**Review by andersy005 — APPROVED** [review]

**Comment by Illviljan** [comment]
Thanks for the reviews! Merging now.

**Pull request merged**